app/assets/src/components/visualizations/table/DataTable.jsx

```diff
@@ -7,24 +7,27 @@ class DataTable extends React.Component {
   constructor(props) {
     super(props);
 
-    this.originalData = this.prepareData(this.props.data);
-
     this.state = {
       filter: "",
-      selectedRows: new Set(this.props.selectedRows || [])
+      selectedRows: new Set(this.props.selectedRows || []),
+      originalData: this.props.data,
+      indexedData: DataTable.indexData(this.props.data)
     };
 
     this.handleCheckBoxChange = this.handleCheckBoxChange.bind(this);
   }
 
   static getDerivedStateFromProps(props, state) {
     let newFilter = DataTable.prepareFilter(props.filter);
+    let newState = {};
     if (newFilter !== state.filter) {
-      return {
-        filter: newFilter
-      };
+      newState.filter = newFilter;
+    }
+    if (props.data !== state.originalData) {
+      newState.originalData = props.data;
+      newState.indexedData = DataTable.indexData(props.data);
```

Comment:
We probably need to reset `selectedRows` too. Maybe to `props.selectedRows` again. Wdyt?

Reply:
added